src/Installer/dotnetup/WindowsPathHelper.cs

```diff
@@ -0,0 +1,443 @@
+    /// <param name="expandedPath">The expanded PATH string to use for detection.</param>
+    /// <returns>The modified unexpanded PATH string.</returns>
+    public static string RemoveProgramFilesDotnetFromPath(string unexpandedPath, string expandedPath)
+    {
+        // Find indices to remove using the expanded path
+        var expandedEntries = SplitPath(expandedPath);
+        var programFilesDotnetPaths = GetProgramFilesDotnetPaths();
+        var indicesToRemove = FindDotnetPathIndices(expandedEntries, programFilesDotnetPaths);
+
+        // Remove those indices from the unexpanded path
+        return RemovePathEntriesByIndices(unexpandedPath, indicesToRemove);
+    }
+
+    /// <summary>
+    /// Removes the Program Files dotnet path from the admin PATH while preserving unexpanded environment variables.
+    /// </summary>
+    /// <returns>The modified unexpanded PATH string.</returns>
+    public static string RemoveProgramFilesDotnetFromAdminPath()
+    {
+        // Read both expanded and unexpanded versions
+        string expandedPath = ReadAdminPath(expand: true);
+        string unexpandedPath = ReadAdminPath(expand: false);
+
+        return RemoveProgramFilesDotnetFromPath(unexpandedPath, expandedPath);
+    }
+
+    /// <summary>
+    /// Adds the Program Files dotnet path to the given PATH string if it's not already present.
+    /// </summary>
+    public static string AddProgramFilesDotnetToPath(string path)
+    {
+        var pathEntries = SplitPath(path);
+        var programFilesDotnetPaths = GetProgramFilesDotnetPaths();
+
+        // Get the primary Program Files dotnet path (non-x86)
+        string primaryDotnetPath = programFilesDotnetPaths.FirstOrDefault() ?? string.Empty;
+        if (string.IsNullOrEmpty(primaryDotnetPath))
+        {
+            return path;
+        }
+
+        // Check if any Program Files dotnet path is already in PATH
+        bool alreadyExists = PathContainsDotnet(pathEntries, programFilesDotnetPaths);
+
+        if (!alreadyExists)
+        {
+            pathEntries.Insert(0, primaryDotnetPath);
+        }
+
+        return string.Join(';', pathEntries);
+    }
+
+    /// <summary>
+    /// Checks if the admin PATH contains the Program Files dotnet path.
+    /// Uses the expanded PATH for accurate detection.
+    /// </summary>
+    public static bool AdminPathContainsProgramFilesDotnet()
+    {
+        var adminPath = ReadAdminPath(expand: true);
+        var pathEntries = SplitPath(adminPath);
+        var programFilesDotnetPaths = GetProgramFilesDotnetPaths();
+
+        return PathContainsDotnet(pathEntries, programFilesDotnetPaths);
+    }
+
+    /// <summary>
+    /// Removes the Program Files dotnet path from the admin PATH.
+    /// This is the main orchestrating method that should be called by commands.
+    /// </summary>
+    /// <returns>0 on success, 1 on failure.</returns>
+    public int RemoveDotnetFromAdminPath()
```

Comment:
Remove the console output from this method and from AddDotnetToAdminPath (except in the case of logging a failure to Console.Error).